activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/DeploymentQueryImpl.java

```diff
@@ -197,6 +198,25 @@ public long executeCount(CommandContext commandContext) {
     return commandContext.getDeploymentEntityManager().findDeploymentsByQueryCriteria(this, page);
   }
 
+  @Override
+  public Deployment executeSingleResult(CommandContext commandContext){
+
+    Deployment deployment = commandContext.getDeploymentEntityManager().selectLatestDeployment("SpringAutoDeployment");
+
+    if (deployment != null){
+      return deployment;
+    } else {
+      List<Deployment> results = executeList(commandContext, null);
+      if (results.size() == 1) {
+        return results.get(0);
+      } else if (results.size() > 1) {
+        throw new ActivitiException("Query return " + results.size() + " results instead of max 1");
+      }
+      return null;
```

Comment:
not proud of this, just mimicking the behavior of the overridden method
without it, all sorts of history tests break
fixing 10-year-old code is not my idea of fun

Reply:
No words for that.